### ISSUE TITLE
Non-redundant nil check

### ISSUE THREAD

**Issue by ryuichis** [oclint/oclint#229]
```
Consider this code:
1. foo  = [some thing];
2. if(foo && [foo check:[bar methodWithSideEffect]]){ };
3. if([foo check:[bar methodWithSideEffect]]){ };

OCLint complains that line 2 has a redundant nil-check and does not complain about line 3. However, due to short-circuit operation,  line 2 fails to invoke '[bar methodWithSideEffect]'  while line 3 does invoke it, even if foo is nil.

These checks are not redundant if any parameter of the method call contains a call to any other method.
```

The discussion of this issue can be found at https://groups.google.com/forum/#!topic/oclint-users/FDmmdAXl_AE

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.
